persistence/src/test/java/com/epam/esm/persistence/repository/impl/GiftCertificateRepositoryImplTest.java

```diff
@@ -0,0 +1,126 @@
+package com.epam.esm.persistence.repository.impl;
+
+import com.epam.esm.persistence.config.TestConfiguration;
+import com.epam.esm.persistence.entity.GiftCertificate;
+import com.epam.esm.persistence.model.SearchSpecification;
+import com.epam.esm.persistence.model.SortSpecification;
+import com.epam.esm.persistence.model.enums.SortDirection;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {TestConfiguration.class})
+@ActiveProfiles("test")
+@Transactional
+class GiftCertificateRepositoryImplTest {
+    private static final Long EXPECTED_SAVED_ID = 5L;
+    private static final int EXPECTED_AMOUNT = 1;
+    private static final LocalDateTime DATE = LocalDateTime.parse("2021-03-25T00:00:00");
+    private static final String SPA = "spa";
+    private static final String FAMILY_CERTIFICATE = "family certificate";
+    private static final String GYM = "gym";
+    private static final String GYM_CERTIFICATE = "for boss of the gym";
+    private static final String POOL = "pool";
+    private static final String CONNECTION_POOL = "for better connection";
+    private static final String CLUP = "clup";
+    private static final String CLUP_GIFT = "for leatherman";
+
+    private static final GiftCertificate FOR_SAVING =
+            new GiftCertificate(null, DATE, DATE, SPA, FAMILY_CERTIFICATE, new BigDecimal(754), Short.valueOf("3"), null);
```

Comment:
with the help of builders it will not be needed to pass null values to constructors